src/libraries/System.Private.CoreLib/src/System/Number.NumberToFloatingPointBits.cs

```diff
@@ -326,16 +1027,38 @@ private static ulong DigitsToUInt64(byte* p, int count)
             Debug.Assert((1 <= count) && (count <= 19));
 
             byte* end = (p + count);
-            ulong res = (ulong)(p[0] - '0');
+            ulong res = 0;
+
+            // parse batches of 8 digits with SWAR
+            while (end - p >= 8)
+            {
+                res = (res * 100000000) + ParseEightDigitsUnrolled(p);
+                p += 8;
+            }
 
-            for (p++; p < end; p++)
+            while (p != end)
             {
                 res = (10 * res) + p[0] - '0';
+                ++p;
             }
 
             return res;
         }
 
+        /// <summary>Parse eight consecutive digits using SWAR</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static uint ParseEightDigitsUnrolled(byte* chars)
+        {
+            ulong val = Unsafe.ReadUnaligned<ulong>(chars);
+            const ulong mask = 0x000000FF000000FF;
+            const ulong mul1 = 0x000F424000000064; // 100 + (1000000ULL << 32)
+            const ulong mul2 = 0x0000271000000001; // 1 + (10000ULL << 32)
+            val -= 0x3030303030303030;
+            val = (val * 10) + (val >> 8); // val = (val * 2561) >> 8;
+            val = (((val & mask) * mul1) + (((val >> 16) & mask) * mul2)) >> 32;
+            return (uint)val;
```

Comment:
This method deserves a documentation comment explaining what its doing.

The average reader or maintainer is going to come back to this and have to work through the algorithm themselves otherwise.

Reply:
i added some extra explainations